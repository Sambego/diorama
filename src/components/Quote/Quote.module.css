.quote {
  margin: 0 auto;
  max-width: 80vw;

  font-weight: 700;
  font-size: 5vw;

  &::before {
    content: '“';
  }

  &::after {
    content: '”';
  }
}

.cite {
  margin: 1rem auto 0 auto;
  padding-right: 5rem;
  max-width: 80vw;
  font-size: 3vw;
  text-align: right;

  &::before {
    content: '-';
  }

  &::after {
    content: '-';
  }
}

@media (min-width: 600px) {
  .cite {
    margin: 1rem auto 0 auto;
    padding-right: 5rem;
  }
}
