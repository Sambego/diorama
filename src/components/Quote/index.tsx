import React, { CSSProperties, PropsWithChildren, ReactNode } from "react";
import PropTypes from "prop-types";

import styles from "./Quote.module.css";

interface QuoteProps {
	className?: string;
	style?: CSSProperties;
	quotee?: ReactNode;
}
function Quote({
	children,
	quotee,
	style,
	className,
}: PropsWithChildren<QuoteProps>) {
	return (
		<>
			<blockquote
				style={style}
				className={`${styles.quote} diorama-quote ${className}`}
			>
				{children}
			</blockquote>
			{quotee && (
				<cite className={`${styles.cite} diorama-cite`}>{quotee}</cite>
			)}
		</>
	);
}

Quote.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	style: PropTypes.shape({}),
	quotee: PropTypes.node,
};

Quote.defaultProps = {
	className: "",
	style: {},
	quotee: undefined,
};

export default Quote;
