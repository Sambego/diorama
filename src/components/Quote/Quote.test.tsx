import React from "react";
import renderer from "react-test-renderer";
import Quote from ".";

it("renders correctly", () => {
	const tree = renderer
		.create(<Quote quotee="Sam Bellen">Nice quote</Quote>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
