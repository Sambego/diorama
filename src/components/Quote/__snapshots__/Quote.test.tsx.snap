// Vitest Snapshot v1

exports[`renders correctly 1`] = `
[
  <blockquote
    className="_quote_1hgwv_1 diorama-quote "
    style={{}}
  >
    Nice quote
  </blockquote>,
  <cite
    className="_cite_1hgwv_17 diorama-cite"
  >
    Sam Bellen
  </cite>,
]
`;
